AsyncDisplayKit/Layout/ASCenterLayoutSpec.mm

```diff
@@ -44,61 +49,52 @@ - (void)setCenteringOptions:(ASCenterLayoutSpecCenteringOptions)centeringOptions
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
   _centeringOptions = centeringOptions;
+  
+  [_internalLayoutSpec setHorizontalPosition:[self horizontalPositionFromCenteringOptions:centeringOptions]];
+  [_internalLayoutSpec setVerticalPosition:[self verticalPositionFromCenteringOptions:centeringOptions]];
 }
 
 - (void)setSizingOptions:(ASCenterLayoutSpecSizingOptions)sizingOptions
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
   _sizingOptions = sizingOptions;
+
+  [_internalLayoutSpec setSizingOption:sizingOptions];
 }
 
 - (ASLayout *)measureWithSizeRange:(ASSizeRange)constrainedSize
 {
-  CGSize size = {
-    constrainedSize.max.width,
-    constrainedSize.max.height
-  };
-
-  // Layout the child
-  const CGSize minChildSize = {
-    (_centeringOptions & ASCenterLayoutSpecCenteringX) != 0 ? 0 : constrainedSize.min.width,
-    (_centeringOptions & ASCenterLayoutSpecCenteringY) != 0 ? 0 : constrainedSize.min.height,
-  };
-  ASLayout *sublayout = [self.child measureWithSizeRange:ASSizeRangeMake(minChildSize, constrainedSize.max)];
-
-  // If we have an undetermined height or width, use the child size to define the layout
-  // size
-  size = ASSizeRangeClamp(constrainedSize, {
-    isnan(size.width) ? sublayout.size.width : size.width,
-    isnan(size.height) ? sublayout.size.height : size.height
-  });
-
-  // If minimum size options are set, attempt to shrink the size to the size of the child
-  size = ASSizeRangeClamp(constrainedSize, {
-    MIN(size.width, (_sizingOptions & ASCenterLayoutSpecSizingOptionMinimumX) != 0 ? sublayout.size.width : size.width),
-    MIN(size.height, (_sizingOptions & ASCenterLayoutSpecSizingOptionMinimumY) != 0 ? sublayout.size.height : size.height)
-  });
-
-  // Compute the centered postion for the child
-  BOOL shouldCenterAlongX = (_centeringOptions & ASCenterLayoutSpecCenteringX);
-  BOOL shouldCenterAlongY = (_centeringOptions & ASCenterLayoutSpecCenteringY);
-  sublayout.position = {
-    ASRoundPixelValue(shouldCenterAlongX ? (size.width - sublayout.size.width) * 0.5f : 0),
-    ASRoundPixelValue(shouldCenterAlongY ? (size.height - sublayout.size.height) * 0.5f : 0)
-  };
-
-  return [ASLayout layoutWithLayoutableObject:self size:size sublayouts:@[sublayout]];
+  return [_internalLayoutSpec measureWithSizeRange:constrainedSize];
 }
 
 - (void)setChildren:(NSArray *)children
```

Comment:
I wonder if we need to override anything else, like -finalLayoutable, to return _internalLayoutSpec?  This is a primary reason to make sure we can't subclass, or find a way to enable that.

One thing I have done before is add #define [deprecated-ish name] [new-superset-functionality-name].  This might allow us to make the property type say it is Center, and the values say that too, but actually be the Relative values under the hood.